Tests/StoreKitUnitTests/PurchasesOrchestratorSK1Tests.swift

```diff
@@ -57,32 +58,147 @@ class PurchasesOrchestratorSK1Tests: PurchasesOrchestratorTestCase {
         expect(self.receiptFetcher.receiptDataReceivedRefreshPolicy) == .always
 
         expect(self.backend.invokedPostReceiptDataCount) == 1
+        expect(self.backend.invokedPostReceiptData).to(beTrue())
+        expect(self.backend.invokedPostReceiptDataParameters?.data) == .receipt(self.receiptFetcher.mockReceiptData)
         expect(self.backend.invokedPostReceiptDataParameters?.productData).toNot(beNil())
         expect(self.backend.invokedPostReceiptDataParameters?.transactionData.presentedOfferingID) == "offering"
         expect(self.backend.invokedPostReceiptDataParameters?.transactionData.source.initiationSource) == .purchase
     }
 
-    func testSK1PurchaseDoesNotAlwaysRefreshReceiptInProduction() async throws {
-        self.customerInfoManager.stubbedCachedCustomerInfoResult = self.mockCustomerInfo
-        self.backend.stubbedPostReceiptResult = .success(self.mockCustomerInfo)
+    func testPurchaseReturnsCorrectValues() async throws {
+        backend.stubbedPostReceiptResult = .success(mockCustomerInfo)
 
-        self.systemInfo.stubbedIsSandbox = false
+        let product = try await self.fetchSk1Product()
+        let payment = storeKit1Wrapper.payment(with: product)
+
+        let (transaction, customerInfo, error, userCancelled) = await withCheckedContinuation { continuation in
+            orchestrator.purchase(sk1Product: product,
+                                  payment: payment,
+                                  package: nil,
+                                  wrapper: self.storeKit1Wrapper) { transaction, customerInfo, error, userCancelled in
+                continuation.resume(returning: (transaction, customerInfo, error, userCancelled))
+            }
+        }
+
+        expect(transaction?.sk1Transaction?.productIdentifier) == product.productIdentifier
+        expect(userCancelled) == false
+        expect(error).to(beNil())
+
+        let expectedCustomerInfo: CustomerInfo = .emptyInfo
+        expect(customerInfo) == expectedCustomerInfo
+    }
+
+    func testPurchaseDoesNotPostReceiptIfPurchaseFailed() async throws {
+        storeKit1Wrapper.mockAddPaymentTransactionState = .failed
+        storeKit1Wrapper.mockTransactionError = NSError(domain: SKErrorDomain,
+                                                        code: SKError.Code.paymentCancelled.rawValue)
```

Comment:
Added this little helper `mockTransactionError` that helps us mock errors in SK1